stdlib/socket/src/main/ballerina/socket/Module.md

```diff
@@ -0,0 +1,63 @@
+## Module overview
+This module provides an implementation for connecting to remote socket server or act as a server for incoming socket request. The module facilitates two types of endpoints as `Client` and `Listener`.
+## Samples
+### Listener endpoints
+The sample given below shows how a listener is used to listen to the incoming socket request. The `onAccept(socket:Caller)` resource function gets invoked when a new client connected. New client will be represented using the `socket:Caller`.
+`onReadReady(socket:Caller, byte[])` resource will get invoked once the remote client sends some data and `onClose(socket:Caller)` will invoke once the client is departed. 
+```ballerina
+import ballerina/io;
+import ballerina/socket;
+
+listener socket:Server server = new ({ port:61598 });
+
+service echoServer on server {
+    resource function onAccept (socket:Caller caller) {
+        io:println("Join: ", caller.remotePort);
+    }
+
+    resource function onReadReady (socket:Caller caller, byte[] content) {
+        _ = caller->write(content);
+    }
+
+    resource function onClose(socket:Caller caller) {
+        io:println("Leave: " + caller.remotePort);
+    }
+
+    resource function onError(socket:Caller caller, error er) {
+        io:println(er.reason());
+    }
+}
+```
+### Client endpoints
+Client endpoints are used to connect to and interact with a socket server. Data can be sent only using the client. Client callbackService needs to retrieve the data and do multiple request/response between client and the server.
```

Comment:
Let's write `callbackService` within single back ticks. 

Also let's change "multiple request/response between client and the server" to "multiple requests/responses between the client and the server".